### ISSUE TITLE
Comparison: Solid vs Ethereum Dapps, Swarm, IPFS, Dat, Storj, Hedera Hashgraph, Internet of People, Safe Network

### ISSUE THREAD

**Issue by niutech** [solid/solid#183]
I am new to Solid but the web page does not describe its features clearly enough. How does Solid compare with existing distributed web platforms, such as Ethereum Dapps, Swarm, IPFS, Dat, Storj, Hedera Hashgraph, Internet of People, Safe Network, etc.? What are the pros and cons of your solution?

(Sorry that I am asking here, but I cannot find an official Solid forum.)

**Comment by bblfish** [comment]
If one were to try to keep track of all the different proposals one would spend one's life doing that and not getting anything done. So your question has to be an open one. I'll answer on a couple I think I can give an answer on

*Ethereum* like most blockchain solutions is distributed data not decentralised. In a way it is a central database that every one has a copy of, and that requires an expensive consensus protocol that a lot of people have to be part of to identify the next version. SoliD allows complete decentralisation without requiring duplication or global consensus. I argue in [Epistemology in the Cloud](https://medium.com/@bblfish/epistemology-in-the-cloud-472fad4c8282) why that is an epistemological requirement on agency.

IPFS is also decentralised in that way but it does not require global data consensus. Everyone can say what they want on IPFS, it's just that everybody could end up knowing it. 

IPFS could be useful and integrated with SoliD for use cases that are happy to have the dara replicated as widely as possible. Given that SoLiD is based on the RDF, ie logic applied to the web, then it can integrate with all kinds of protocols. SoLiD starts with http(s) as it is the most widely distributed protocol in existence.

**Comment by melvincarvalho** [comment]
@niutech short answer is that solid, for the most part, is a semantic layer that sits on top of those other technologies, and should be complimentary.  Lots more details of course, but I thought Id give a high level comment.

**Comment by melvincarvalho** [comment]
> Thank you for explaining it. But how does Solid decentralization work compared with e.g. IPFS? When I upload my data to a Solid pod and it goes down, isn't it a single point of failure, unlike distributed IPFS nodes?

It certainly is, unless we make some content addressable backup space (which is quite doable btw).

But the alternative is what?  To have redundancy in IPFS you need to pin a file, which is a paid service.  And only as long as the pins survive.  They are basically different cost models, with different cost / benefit / risk profiles.

I'd really like to work on creating multihash content hosted on solid pods.  I dont think it's hard -- just put the file in .well-known/multihash/hash and get others to mirror it with some routing and a key space.  Fun weekend project, if I find time! :)

**Comment by imta11** [comment]
I signed up for the pod. When will the solid pod come in the mail?

**Comment by timathom** [comment]
> But the alternative is what? To have redundancy in IPFS you need to pin a file, which is a paid service. And only as long as the pins survive. They are basically different cost models, with different cost / benefit / risk profiles.

@melvincarvalho Although there may be services that you can pay to pin your files in IPFS for you, pinning itself--as far as I understand--is a core feature of IPFS, something that anyone can do. Do you have a source for saying it's a paid service?

**Comment by melvincarvalho** [comment]
@timathom you're right.  It doesnt have to be a pinned service.  You can pin it on your laptop and access it from your laptop (if you would want to do such a thing), or you can pin it on your laptop and access from another device, but then you have to keep your laptop on, and connected to the net.  Im trying to get at the point (probably badly phrased) that there is a cost model for the redundancy.

**Comment by niutech** [comment]
@melvincarvalho what if you put Solid pods on top of something like IPFS, so that a pod is an IPFS node as well as an HTTPS gateway? Wouldn't it provide the best of both worlds?

**Comment by melvincarvalho** [comment]
On Thu, 4 Oct 2018 at 02:08, niu tech <notifications@github.com> wrote:

> @melvincarvalho <https://github.com/melvincarvalho> what if you put Solid
> pods on top of something like IPFS, so that a pod is an IPFS node as well
> as an HTTPS gateway? Wouldn't it provide the best of both worlds?
>

Yes, that's the idea.  But does IPFS have HTTP headers?

**Comment by niutech** [comment]
@melvincarvalho This is discussed in https://github.com/ipfs/ipfs/issues/75 and .

**Comment by kggp1995** [comment]
> Im trying to get at the point (probably badly phrased) that there is a cost model for the redundancy.

I think a cost model to help users manage / address redundancy makes sense.

In case of IPFS, we can think of encrypting data & then pinning them in the IPFS redundancy pool. This makes the data management fairly distributed & safe from unauthorized access at the same time.

Crypto payments to the interested parties in persisting these redundant shards of info is a common concept that could be availed as an option.

@timbl @melvincarvalho I hope this concept is in-line to the vision of Solid? Just checking..

**Comment by kggp1995** [comment]
@melvincarvalho glad to know! I will fork the server repo & get started on the IPFS Service abstraction.

Will ping you if I hit a tech roadblock.

**Comment by thecryptodrive** [comment]
From what I understand a Solid Pod is essentially an encrypted container for your online identity and data, it does not exist in any distributed context and as mentioned above is a single point of failure and hence, once located, is easily censored as well. A more robust network could be created by treating Pods as nodes for sharding purposes, you would still need a Pod to join the network and house your data but it also adds to the global sharding pool, where encrypted shards of data from x other pods are stored within your Pod and in turn your data is sharded to other Pods. 

This basically creates a self-healing network where you could essentially rebuild your Pod from the shard network if the host server ever failed. Timestamped version restore capabilities would also be useful, should your shard data ever become corrupted or compromised, allowing rollback to a previous state archived by the shard mesh network. 

Basically a Pod as a single container is a nice idea as a starting point, but with sharding, each new Pod that joins would also add to the security and immutability of the Solid network community and make it a truely robust and technically marvelous concept.

**Comment by megoth** [comment]
For those that want to continue this discussion, I can recommend trying out the new forum! ^_^ https://forum.solidproject.org/

**Comment by melvincarvalho** [comment]
@megoth thanks for the pointer.  Just a point on process.

As we dont yet have a process relating the discourse site to github (which is our primary issue system), could I suggest the following.

If there is consensus of moving an issue from github to discourse, create a pointer to the previous issue there, and vice versa.

This topic has generated some interested and is asked about frequently.  I think it would be a good candidate for an FAQ.  Consider reopening temporarily.

**Comment by megoth** [comment]
> If there is consensus of moving an issue from github to discourse, create a pointer to the previous issue there, and vice versa.

There is a thread started from this issue, [Distributed Solid (based on IPFS, SAFE Network, etc.)](https://forum.solidproject.org/t/distributed-solid-based-on-ipfs-safe-network-etc/79), although I would argue the thread is slightly more specific than the questions initially proposed here.

In general I think people should link to this issue if they have threads inspired from it, and add a comment here pointing to the new thread, in case people are following this issue. (I guess this comment ensures that for the aforementioned thread.)

> This topic has generated some interested and is asked about frequently. I think it would be a good candidate for an FAQ. Consider reopening temporarily.

This issue and the Discourse thread are good resources for keeping track of the interest in this topic. It might be a lot of threads going forward, and could be a bit confusing for newcomers to get into, but hey, that's threaded discussions for you ;)

We could reopen this issue to make it easier to find? But could mark it with a specific label that conveys to us developers that this is merely a discussion issue?
